File: 0162-find-peak-element/0162-find-peak-element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int n = nums.size();
        for (int i = 0; i < n; i++) {
            int left, right;
            if (i == 0) left = INT_MIN;
            else left = nums[i - 1];
            if (i == n - 1) right = INT_MIN;
            else right = nums[i + 1];
            if (nums[i] > left && nums[i] > right) return i;
        }
        return 0;
    }
};

File: 0162-find-peak-element/0162-find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        for i in range(len(nums)):
            if i == 0:
                left = float('-inf')
            else:
                left = nums[i-1]
            if i == len(nums)-1 :
                right = float('-inf')
            else:
                right = nums[i+1]
            if left < nums[i] > right:
                return i
